apps/ledger-live-desktop/src/renderer/families/ethereum/MaxBaseFeeField.jsx

```diff
@@ -0,0 +1,142 @@
+// @flow
+import React, { useCallback } from "react";
+import { BigNumber } from "bignumber.js";
+import invariant from "invariant";
+import styled from "styled-components";
+import { withTranslation, Trans } from "react-i18next";
+import type { TFunction } from "react-i18next";
+import type { Account, TransactionStatus } from "@ledgerhq/live-common/types/index";
+import type { Transaction } from "@ledgerhq/live-common/families/ethereum/types";
+import { formatCurrencyUnit } from "@ledgerhq/live-common/currencies/index";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { inferDynamicRange } from "@ledgerhq/live-common/range";
+import Label from "~/renderer/components/Label";
+import FeeSliderField from "~/renderer/components/FeeSliderField";
+import InputCurrency from "~/renderer/components/InputCurrency";
+import Box from "~/renderer/components/Box";
+import LabelWithExternalIcon from "~/renderer/components/LabelWithExternalIcon";
+import { openURL } from "~/renderer/linking";
+import { track } from "~/renderer/analytics/segment";
+import { urls } from "~/config/urls";
+import TranslatedError from "~/renderer/components/TranslatedError";
+
+const ErrorContainer = styled(Box)`
+  margin-top: 0px;
+  font-size: 10px;
+  width: 100%;
+  transition: all 0.4s ease-in-out;
+  will-change: max-height;
+  max-height: ${p => (p.hasError ? 60 : 0)}px;
+  min-height: ${p => (p.hasError ? 22 : 0)}px;
+  `;
+
+  
+const ErrorDisplay = styled(Box)`
+  color: ${p => p.theme.colors.pearl};
+`;
+
+const WarningDisplay = styled(Box)`
+  color: ${p => p.theme.colors.warning};
+`;
+
+type Props = {
+  account: Account,
+  transaction: Transaction,
+  status: TransactionStatus,
+  updateTransaction: (updater: any) => void,
+};
+
+let lastNetworkNextBaseFeePerGas;
```

Comment:
not super fan of the local variable living in a component, what was the intended behavior here? I wonder if this should be a settings, because i'm not sure about the intended "lifecycle" of this memory feature.
regardless of what the behavior is, this would be better to have it more on the "live-common" side because I imagine it would be shared with moblie too and is more related to the coin implementation than the UI?

Reply:
TBH I was basing it on an already present piece of code for handling GasPrice (https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/families/ethereum/GasPriceField.jsx#L20). Thanks for the comment, that makes sense, the local variable shouldn't be used here. I changed it to using useMemo(), I also couldn't observe the blinks, so I think that at least the behaviour should be correct.